nomad/state/state_store.go

```diff
@@ -5694,6 +5797,11 @@ func (s *StateStore) updateSummaryWithAlloc(index uint64, alloc *structs.Allocat
 		return fmt.Errorf("unable to find task group in the job summary: %v", alloc.TaskGroup)
 	}
 
+	vtgSummary, ok := jobSummary.VersionedSummary[alloc.Job.Version]
+	if !ok {
+		return fmt.Errorf("unable to find versioned summary for job version: %v", alloc.Job.Version)
+	}
+
 	summaryChanged := false
 	if existingAlloc == nil {
```

Comment:
What should I be trying to do with a new alloc in terms of versioned summary? If it has no version, what would I increment/decrement? In practice, my clientStatus counts for all allocs in the "versioned" summary seem to match the non-versioned one, so I'm not sure if anything is warranted here but wanted to note it.

(I don't think we need to mess with "queued" in the versioned summary.)